packages/compiler/src/renderers/components/ImageView/index.tsx

```diff
@@ -0,0 +1,119 @@
+/******************************************************************
+ * created by lqy at 06/16/2021
+ ******************************************************************/
+
+import React from 'react';
+import Modal from '@material-ui/core/Modal';
+import Fade from '@material-ui/core/Fade';
+import { makeStyles, createStyles } from '@material-ui/styles';
+
+export interface PreviewProps {
+    children?: React.ReactElement;
+}
+
+type ChildrenProps = React.DOMAttributes<Element> & {
+    ref: React.RefObject<HTMLImageElement>;
+    style: React.CSSProperties;
+} ;
+
+const useStyles = makeStyles( () => {
+    return createStyles( {
+        viewWrapper : {
+            border : 'none',
+            position : 'absolute',
+            left : 0,
+            right : 0,
+            top : 0,
+            bottom : 0,
+            overflow : 'auto'
+        },
+        img : {
+            transition : 'all 0.6s ease-in-out',
+            cursor : 'zoom-out'
+        }
+    } );
+} );
+
+export default function ImageView( props: PreviewProps ): JSX.Element {
+    const { children } = props;
+    const styles = useStyles( props );
+    const [ open, setOpen ] = React.useState( false );
+
+    const handleShow = (): void => {
+        setOpen( true );
+    };
+
+    const handleClose = (): void => {
+        setOpen( false );
+    };
+
+    const wraperChildren = (): React.ReactElement => {
+        const newProps: ChildrenProps = {
```

Comment:
The `newProps` variable is only used in the `newChildren` function, so consider to just declare the `imageRef` here, and just declare this variable in the function which is using it.

```suggestion
        const imageRef = React.createRef<HTMLImageElement>();
```